network/src/config.rs

```diff
@@ -50,56 +57,71 @@ impl Config {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
         node_id: Option<u64>,
+        node_type: NodeType,
         desired_address: SocketAddr,
         minimum_number_of_connected_peers: u16,
         maximum_number_of_connected_peers: u16,
-        bootnodes_addresses: Vec<String>,
-        is_bootnode: bool,
-        is_crawler: bool,
+        beacon_addresses: Vec<String>,
+        sync_provider_addresses: Vec<String>,
         peer_sync_interval: Duration,
     ) -> Result<Self, NetworkError> {
-        // Convert the given bootnodes into socket addresses.
-        let mut bootnodes = Vec::with_capacity(bootnodes_addresses.len());
-        for bootnode_address in bootnodes_addresses.iter() {
-            if let Ok(bootnode) = bootnode_address.parse::<SocketAddr>() {
-                bootnodes.push(bootnode);
-            }
-        }
+        // Convert the given seeded nodes into socket addresses.
+        let beacons: Vec<SocketAddr> = beacon_addresses
+            .into_iter()
+            .flat_map(|addr| addr.parse().ok())
```

Comment:
```suggestion
            .filter_map(|addr| addr.parse().ok())
```
it conveys the logic a bit better